R/recall_data.R

```diff
@@ -191,28 +200,41 @@ recall_details <- function(recall_number, limit = 25) {
             call. = FALSE)
     }
 
-    compiled_df <- vector(mode = "list", length = length(recall_number))
+
     i <- 1
     for (single_number in recall_number) {
 
+        #initialize dataframe
+        if (i == 1){
+
+            # format the url string
+            url_ <- "https://vrdb-tc-apicast-production.api.canada.ca/eng/vehicle-recall-database/v1/recall-summary/recall-number/"
+            url_ <- paste(url_, toString(single_number), sep = "")
+
+            # api call, returns class vrd_api
+            api_output <- call_vrd_api(url_, single_number)
+
+            #create dataframe
+            compiled_df <- clean_vrd_api(api_output)
+        }else{
+
         # format the url string
         url_ <- "https://vrdb-tc-apicast-production.api.canada.ca/eng/vehicle-recall-database/v1/recall-summary/recall-number/"
         url_ <- paste(url_, toString(single_number), sep = "")
 
         # api call, returns class vrd_api
-        api_output <- call_vrd_api(url_, single_number, limit)
+        api_output <- call_vrd_api(url_, single_number)
 
-        # convert content to a dataframe
-        ### NOTE that it may make sense to do format differently here
-        compiled_df[i] <- api_output$content
+        # add content to a dataframe
+        compiled_df<-rbind(compiled_df, clean_vrd_api(api_output))
+        }
 
         Sys.sleep(1)
```

Comment:
Sleeping for 1 second?

you found this to be necessary?

Reply:
that's from my previous PR - only allowed to make 60 calls/min so this throttles it to be at that

Reply:
👍